src/app/wallets/update-on-chain-receipt.ts

```diff
@@ -129,10 +129,10 @@ const processTxForWallet = async (
       return recorded
     }
 
-    const account = await AccountsRepository().findById(wallet.accountId)
-    if (account instanceof Error) return account
-
     if (!recorded) {
+      const account = await AccountsRepository().findById(wallet.accountId)
```

Comment:
@dolcalmi did we want this account existence check for some reason outside the `if (!recorded)` block?